spec/js/query_params_spec.js

```diff
@@ -0,0 +1,42 @@
+import { describe, it } from 'node:test';
+import assert from 'node:assert';
+import { JSDOM } from 'jsdom';
+import { storeUrlQueryParams, removeUrlQueryParams } from '../../assets/js/query_params.js';
+
+function setupTestEnvironment(url) {
+  const dom = new JSDOM('', { url });
+  global.window = dom.window;
+  global.localStorage = dom.window.localStorage;
+  global.URLSearchParams = dom.window.URLSearchParams;
+  return dom;
```

Comment:
Should we tear this down at some point, especially since it's assigning to `global` and will bleed over across tests?

Reply:
Good call. I'll spin this out to a separate ticket.